torch/onnx/verification.py

```
@@ -771,3 +789,835 @@ def verify(
+        graph_outputs = list(self.graph.outputs())
+        assert len(graph_outputs) == len(
+            pt_outs
+        ), f"{len(graph_outputs)} vs {len(pt_outs)}\nGraph: {self.graph}"
+        return {v.debugName(): o for v, o in zip(graph_outputs, pt_outs)}
+
+    def _args_and_params_for_partition_graph(
+        self,
+        graph: torch.Graph,
+        bridge_kwargs: Mapping[str, Union[_NumericType, Sequence[_NumericType]]],
+        full_kwargs: Mapping[str, torch.Tensor],
+        full_params: Mapping[str, torch.Tensor],
+    ):
+        input_names = [input.debugName() for input in graph.inputs()]
+        args = tuple(bridge_kwargs[k] for k in input_names if k in bridge_kwargs)
+        args += tuple(full_kwargs[k] for k in input_names if k in full_kwargs)
+        params = {k: full_params[k] for k in input_names if k in full_params}
+        assert len(args) + len(params) == len(
+            input_names
+        ), f"{len(args)} + {len(params)} vs {len(input_names)}: {input_names}"
+        return args, params
+
+    def verify_export(
+        self, options: VerificationOptions
+    ) -> Tuple[
+        Optional[AssertionError],
+        torch.Graph,
+        Union[_NumericType, Sequence[_NumericType]],
+        Union[_NumericType, Sequence[_NumericType]],
+    ]:
+        return verify_aten_graph(
+            self.graph,
+            input_args=self.input_args,
+            params_dict=self.params_dict,
+            export_options=self.export_options,
+            verification_options=options,
+        )
+
+    def find_mismatch(
+        self,
+        options: Optional[VerificationOptions] = None,
+    ):
+        self.clear()
+
+        if options is None:
+            options = VerificationOptions()
+
+        if self.export_options.verbose:
+            print(self.graph)
+
+        if len(list(self.graph.outputs())) == 0:
+            return
+
+        assert len(self.input_args) + len(self.params_dict) == len(
```

Does this happen because of user input? Do we need to raise an error?

---

Not really due to user input. More so on edge cases when graphs are partitioned. Like unexpected extra or omitted inputs.